src/controllers/dataset.ts

```diff
@@ -336,3 +340,80 @@ export const getFactTableDefinition = async (req: Request, res: Response) => {
   res.status(200);
   res.json(factTableDto);
 };
+
+async function addDirectoryToZip(
+  zip: JSZip,
+  datasetFiles: Map<string, FileImportDto>,
+  directory: string,
+  fileService: StorageService
+) {
+  const directoryList = await fileService.listFiles(directory);
+  for (const fileEntry of directoryList) {
+    let filename: string;
+    if ((fileEntry as DataLakeFileEntry).name) {
+      const entry = fileEntry as DataLakeFileEntry;
+      if (entry.isDirectory) {
+        await addDirectoryToZip(zip, datasetFiles, `${directory}/${entry.name}`, fileService);
+        continue;
+      }
+      filename = (fileEntry as DataLakeFileEntry).name;
+    } else {
+      filename = fileEntry as string;
+    }
+    const originalFilename = datasetFiles.get(filename)?.filename || filename;
+    zip.file(originalFilename, await fileService.loadBuffer(filename, directory));
+  }
+}
+
+function collectFiles(dataset: Dataset): Map<string, FileImportDto> {
+  const files: Map<string, FileImportDto> = new Map<string, FileImportDto>();
+  if (dataset.measure.lookupTable) {
+    files.set(dataset.measure.lookupTable.filename, FileImportDto.fromFileImport(dataset.measure.lookupTable));
+  }
+  dataset.dimensions.forEach((dimension) => {
+    if (dimension.lookupTable) {
+      files.set(dimension.lookupTable.filename, FileImportDto.fromFileImport(dimension.lookupTable));
+    }
+  });
+  dataset.revisions.forEach((revision) => {
+    if (revision.dataTable) {
+      files.set(revision.dataTable.filename, FileImportDto.fromFileImport(revision.dataTable));
+    }
+  });
+  return files;
+}
+
+export const getEverythingFromDatalake = async (req: Request, res: Response) => {
```

Comment:
I'd prefer to avoid using the word "datalake" or "blobstorage" outside of their respective storage classes...

We've now got `req.fileService` which might be an instance of blob storage or datalake, and this should ideally (if we want to be able to test this endpoint in CI) work with either.

e.g. `getAllFilesForDataset` / `fullDatasetZip` or similar.

Reply:
Renamed to `getAllFilesForDataset`